testingWithMocha/tests/factorial_test.js

```diff
@@ -0,0 +1,43 @@
+var assert = require('assert');
+var Calculate = require('../factorial.js');
+
+describe('Calculate', () => {
+    describe('.factorial', () => {
+        it('returns factorial of 12!', () => {
+            const expectedResult = 479001600;
+            const input = 12;
+            const result = Calculate.factorial(input);
+            assert.strictEqual(result, expectedResult);
+        });
+        it('returns factorial of 5!', () => {
```

Comment:
nit: always have a linebreak between tests.

I gave a presentation on good tests in a few places, but essentially, tests have 3 main purposes:
1) **To prevent regressions/bugs** - Write your tests in such a way that each test covers a different category of input/case such that if something is out of the ordinary or a dev changes some code by accident, the test catches and warns them that they broke the code in a way that they maybe hadn't expected

2) **Tests as Documentation** - Often the production code isn't enough to really communicate all the business cases that it's meant to handle. But in the tests, you can detail each edge-case and explain what should & should not be happening as a result of X input. This means don't only test happy-paths, but also test the negative cases to clearly define what your code should and should not be doing. And part of this is legibility. It should be clear and easy to peruse your tests, as if they're documents. If there are like a dozen tests all at the same level of indentation, you probably could do with grouping them in a nested `describe` block like chapters.

3) **Actionable on failure** - Sometimes when something breaks I see ALL the tests for a given suite fail at once. This can legitimately happen, but sometimes it's only doing that because all the tests have the same basic set of expectations built into them. No, don't do that. Write one test for the basic assumptions, and then each test should be confirming its own thing. So when something breaks, the tests that fail should be targeted and actionable. You never know if there's a time crunch on whatever happened to fail the test. Can you imagine if every warning light you got on a car was "something's wrong"? No, we have "check engine", "low fuel", "battery low", etc. Targeted messages on failure

Putting all of this together, I'd probably refactor your tests to something like:

```ts
describe('Calculate', () => {
  describe('.factorial', () => {
    describe('handles positive inputs', () => {
      [
        { input: 1, expectedResult: 1 },
        { input: 3, expectedResult: 3 },
        { input: 5, expectedResult: 120 },
        { input: 12, expectedResult: 479001600 },
      ].forEach(({ input, expectedResult }) => {
        it(`like ${input}`, () => {
          assert.strictEqual(Calculate.factorial(input), expectedResult);
        });
      });
      
    describe('handles non-standard inputs', () => {
      it('like 0', () => {
        assert.strictEqual(Calculate.factorial(0), 1);
      });
      
      it('like negative numbers (undefined)', () => {
        assert.strictEqual(Calculate.factorial(-1), undefined);
      });
      
      it('like non-numeric inputs (throws)', () => {
        assert.throws(() => {
          assert.strictEqual(Calculate.factorial('five'), undefined);          
        }, TypeError);
      });
    });
  });
});
```

Notice, as well, that my cascading `describe` & `it` blocks all read like a sentence. Because when they're printed on the output they'll look like:
```
Calculate > .factorial > handles positive inputs > like 1
Calculate > .factorial > handles positive inputs > like 5
...
Calculate > .factorial > handles non-standard inputs > like non-numeric inputs (throws)
```
Which are just so much easier to read/parse than lazier shorthand.

Also notice that in the happy path tests, that I abstracted away a lot of the duplicated code in favor of a loop. I'm not saying to do that all the time, but it helps to recognize a lot of the duplication in your testing and pull that out somehow, because it now makes it clearer (to that "documentation" tenet), about what each test case is doing. Like, what makes it different from each other test case.

And if you make the act of WRITING a new test (or adding/editing cases), you incentivize devs to write more good tests, which is always a good thing.